### PR TITLE
Enable basic auth tests with domainless custom authenticator

### LINKED ISSUE #1111: Basic authentication tests fail in CI when using IIS-hosted services
The Basic authentication tests fail on CI runs against IIS-hosted WCF services but pass in self-hosted.  Likely cause is that Basic Authentication is not enabled for the CI web sites in IIS.
These are the 2 failing tests:
```
Https_ClientCredentialTypeTests.BasicAuthenticationInvalidPwd_throw_MessageSecurityException 
Https_ClientCredentialTypeTests.BasicAuthentication_RoundTrips_Echo
```

Failure looks like this:
```
Https_ClientCredentialTypeTests.BasicAuthenticationInvalidPwd_throw_MessageSecurityException [FAIL]
        Assert.Throws() Failure
        Expected: typeof(System.ServiceModel.Security.MessageSecurityException)
        Actual:   typeof(System.ServiceModel.CommunicationException): The server did not provide a meaningful reply; this might be caused by a contract mismatch, a premature session shutdown or an internal server error.
        Stack Trace:
           D:\j\workspace\outerloop_win---1452b487\src\System.Private.ServiceModel\src\System\ServiceModel\Channels\ServiceChannel.cs(764,0): at System.ServiceModel.Channels.ServiceChannel.Call(String action, Boolean oneway, ProxyOperationRuntime operation, Object[] ins, Object[] outs, TimeSpan timeout)
           D:\j\workspace\outerloop_win---1452b487\src\System.Private.ServiceModel\src\System\ServiceModel\Channels\ServiceChannelProxy.cs(371,0): at System.ServiceModel.Channels.ServiceChannelProxy.InvokeService(MethodCall methodCall, ProxyOperationRuntime operation)
           D:\j\workspace\outerloop_win---1452b487\src\System.Private.ServiceModel\src\System\ServiceModel\Channels\ServiceChannelProxy.cs(136,0): at System.ServiceModel.Channels.ServiceChannelProxy.Invoke(MethodInfo targetMethod, Object[] args)
           --- End of stack trace from previous location where exception was thrown ---
              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
              at System.Reflection.DispatchProxyGenerator.Invoke(Object[] args)
              at generatedProxy_2.Echo(String )
           D:\j\workspace\outerloop_win---1452b487\src\System.Private.ServiceModel\tests\Scenarios\Security\TransportSecurity\Https\ClientCredentialTypeTests.cs(82,0): at Https_ClientCredentialTypeTests.<>c.<BasicAuthenticationInvalidPwd_throw_MessageSecurityException>b__4_0()
     Https_ClientCredentialTypeTests.BasicAuthentication_RoundTrips_Echo [FAIL]
        Test Case: BasicAuthentication FAILED with the following errors: Basic echo test.
  Test variation:...
  BasicAuthentication_RoundTrips_Echo
  Using address: 'https://wcfcoresrv2.cloudapp.net/WcfService15//BasicAuth.svc//https-basic'
        Unexpected exception was caught: System.ServiceModel.CommunicationException: The server did not provide a meaningful reply; this might be caused by a contract mismatch, a premature session shutdown or an internal server error.
           at System.ServiceModel.Channels.ServiceChannel.Call(String action, Boolean oneway, ProxyOperationRuntime operation, Object[] ins, Object[] outs, TimeSpan timeout) in D:\j\workspace\outerloop_win---1452b487\src\System.Private.ServiceModel\src\System\ServiceModel\Channels\ServiceChannel.cs:line 764
           at System.ServiceModel.Channels.ServiceChannelProxy.InvokeService(MethodCall methodCall, ProxyOperationRuntime operation) in D:\j\workspace\outerloop_win---1452b487\src\System.Private.ServiceModel\src\System\ServiceModel\Channels\ServiceChannelProxy.cs:line 371
           at System.ServiceModel.Channels.ServiceChannelProxy.Invoke(MethodInfo targetMethod, Object[] args) in D:\j\workspace\outerloop_win---1452b487\src\System.Private.ServiceModel\src\System\ServiceModel\Channels\ServiceChannelProxy.cs:line 136
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Reflection.DispatchProxyGenerator.Invoke(Object[] args)
           at generatedProxy_2.Echo(String )
           at Https_ClientCredentialTypeTests.BasicAuthentication_RoundTrips_Echo() in D:\j\workspace\outerloop_win---1452b487\src\System.Private.ServiceModel\tests\Scenarios\Security\TransportSecurity\Https\ClientCredentialTypeTests.cs:line 40

```

### PR DESCRIPTION
**Pull request by mconnew**
Fixes #1111

### CODE REVIEW

**Comment by roncain** [comment]
These tests are failing on *nix with the error:
```
MESSAGE:
Test Case: BasicAuthentication FAILED with the following errors: Basic echo test.Test variation:...\nBasicAuthentication_RoundTrips_Echo
Using address: 'https://wcfcoresrv2.cloudapp.net/WcfService4//BasicAuth.svc/https-basic'\nUnexpected exception was caught: System.ServiceModel.CommunicationException: An error occurred while sending the request. ---> System.Net.Http.HttpRequestException: An error occurred while sending the request. ---> System.Net.Http.CurlException: SSL peer certificate or SSH remote key was not OK
at System.Net.Http.CurlHandler.ThrowIfCURLEError(CURLcode error)
 at System.Net.Http.CurlHandler.MultiAgent.FinishRequest(StrongToWeakReference`1 easyWrapper, CURLcode messageResult)\n --- End of inner exception stack trace ---
```

The SSL_Available ConditionalFact was used for all tests that failed with this kind of exception in the past, pending deeper investigation of what is required for automated installation scripts.

File: src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Https/ClientCredentialTypeTests.cs [hunk 1]
```diff
@@ -24,20 +27,18 @@ static Https_ClientCredentialTypeTests()
 #if FULLXUNIT_NOTSUPPORTED
     [Fact]
 #else
-    [ConditionalFact(nameof(Root_Certificate_Installed), nameof(Basic_Authentication_Available))]
+    [ConditionalFact(nameof(Root_Certificate_Installed))]
```
Comment (roncain) [inline src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Https/ClientCredentialTypeTests.cs]: Why did you remove this ConditionalFact?  Do we know Basic_Authentication_Available is true for all platforms, both self-host and IIS-host?

File: src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Https/ClientCredentialTypeTests.cs [hunk 2]
```diff
@@ -9,11 +9,14 @@
 using System.Text;
 using Xunit;
 using Infrastructure.Common;
+using System.ServiceModel.Channels;
```
Comment (roncain) [inline src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Https/ClientCredentialTypeTests.cs]: Nit: usually all System usings are alphabetized above the others.

File: src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Https/ClientCredentialTypeTests.cs [hunk 3]
```diff
@@ -24,20 +27,18 @@ static Https_ClientCredentialTypeTests()
 #if FULLXUNIT_NOTSUPPORTED
     [Fact]
 #else
-    [ConditionalFact(nameof(Root_Certificate_Installed), nameof(Basic_Authentication_Available))]
+    [ConditionalFact(nameof(Root_Certificate_Installed))]
 #endif
     [OuterLoop]
     public static void BasicAuthentication_RoundTrips_Echo()
```
Comment (roncain) [inline src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Https/ClientCredentialTypeTests.cs]: Not required for this PR, but we need some negative tests, especially now that we can control the server-side authentication.  Would you mind opening a new issue to add negative tests for BasicAuth?

**Comment by roncain** [comment]
@StephenBonikowsky is doing a massive sweep to "version-ize" the tests in PR #1366.  I think it would be most efficient if we allowed his PR to be completed and then re-apply the changes in this PR.  My own PeerTrust branch is also waiting for that PR to finish.

**Comment by mconnew** [comment]
@iamjasonp, do you know why this test is getting this exception: `System.Net.Http.CurlException: Peer certificate cannot be authenticated with given CA certificates`

**Comment by iamjasonp** [comment]
I wouldn't expect that to pass -  root certs aren't installed on CI machines. I'm mildly surprised that Root_Certificate_Installed is evaluating to true.

**Comment by iamjasonp** [comment]
Ah, I see - note that CA certificates being used by cURL does not use the x509 stores. So while Root_Certificate_Installed is evaluating to true (because certs are being installed fine in the .NET x509 cert stores), https calls won't work because they depend on the system cert store.

File: src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Https/ClientCredentialTypeTests.cs (hunk 1 above)
Comment (mconnew) [inline src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Https/ClientCredentialTypeTests.cs]: The the whole point of this change, it enables Basic authentication by using a custom authenticator on the server side so you don't need to use real credentials to test it. As there's no such thing as ambient credentials with Basic authentication, there's no need to have a test variant which uses a real domain.

File: src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Https/ClientCredentialTypeTests.cs (hunk 3 above)
Comment (mconnew) [inline src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Https/ClientCredentialTypeTests.cs]: There were two tests for basic auth, and one of them is a negative test and also is passing.

**Comment by iamjasonp** [comment]
https://github.com/dotnet/wcf/issues/1304 will help with getting these tests running in Helix/CI

**Comment by mconnew** [comment]
@dotnet-bot test all outerloop

**Comment by mconnew** [comment]
@dotnet-bot test all outerloop

**Comment by iamjasonp** [comment]
If the default platforms (Windows_NT, Ubuntu14.04, CentOS7.1, OSX) pass, don't worry about the remaining outerloop CI failures for the new platforms we turned on in CI

**Comment by mconnew** [comment]
Windows_NT Debug failure was unrelated so merging.
